engine/src/main/java/org/terasology/engine/persistence/serializers/NetworkEntitySerializer.java

```diff
@@ -130,7 +130,7 @@ private void serializeComponentDelta(Component oldComponent, Component newCompon
                                          boolean componentInitial) {
         ComponentMetadata<?> componentMetadata = componentLibrary.getMetadata(oldComponent.getClass());
         if (componentMetadata == null) {
-            logger.error("Unregistered component type: {}", oldComponent.getClass());
+            logger.atError().addArgument(() -> oldComponent.getClass()).log("Unregistered component type: {}");
```

Comment:
`oldComponent.getClass()` used before in line 131, create and reference local variable